app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,505 @@
+import com.google.samples.apps.sunflower.utilities.InjectorUtils
+import com.google.samples.apps.sunflower.viewmodels.PlantDetailViewModel
+import dev.chrisbanes.accompanist.coil.CoilImageWithCrossfade
+import dev.chrisbanes.accompanist.mdctheme.MaterialThemeFromMdcTheme
+
+private const val ParallaxDelta = 2f
+private const val HeaderTransitionOffset = 150f
+private val SunflowerFabShape =
+    RoundedCornerShape(topLeft = 0.dp, topRight = 30.dp, bottomRight = 0.dp, bottomLeft = 30.dp)
+
+/**
+ * As these callbacks are passed in through multiple Composables, to avoid having to name
+ * parameters to not mix them up, they're aggregated in this class.
+ */
+data class PlantDetailsCallbacks(
+    val onFabClicked: () -> Unit,
+    val onBackClicked: () -> Unit,
+    val onShareClicked: () -> Unit
+)
+
+@Composable
+fun PlantDetailsScreen(
+    plantId: String,
+    onBackClicked: () -> Unit,
+    onShareClicked: (String) -> Unit
+) {
+    // ViewModel and LiveDatas needed to populate the plant details info on the screen
+    val plantDetailsViewModel: PlantDetailViewModel = viewModel(
+        factory = InjectorUtils.providePlantDetailViewModelFactory(ContextAmbient.current, plantId)
+    )
+    val plant = plantDetailsViewModel.plant.observeAsState().value
+    val isPlanted = plantDetailsViewModel.isPlanted.observeAsState().value
+
+    // Every time there's a new value for plant or isPlanted LiveData, this block will get executed
+    if (plant != null && isPlanted != null) {
+        val context = ContextAmbient.current
+        val view = ViewAmbient.current
+
+        PlantDetails(plant, isPlanted, PlantDetailsCallbacks(
+            onBackClicked = onBackClicked,
+            onFabClicked = {
+                plantDetailsViewModel.addPlantToGarden()
+                Snackbar.make(view, R.string.added_plant_to_garden, Snackbar.LENGTH_LONG).show()
+            },
+            onShareClicked = {
+                val shareText = context.resources.getString(R.string.share_text_plant, plant.name)
+                onShareClicked(shareText)
+            }
+        ))
+    }
+}
+
+@VisibleForTesting
+@Composable
+fun PlantDetails(
+    plant: Plant,
+    isPlanted: Boolean,
+    callbacks: PlantDetailsCallbacks,
+    modifier: Modifier = Modifier
+) {
+    // PlantDetails owns the scrollerPosition to simulate CollapsingToolbarLayout's behavior
+    val scrollerPosition = ScrollerPosition()
+    var toolbarState by state { ToolbarState.HIDDEN }
+
+    // Transition that fades in/out the header with the image and the Toolbar
+    Transition(
+        definition = toolbarTransitionDefinition,
+        toState = toolbarState
+    ) { transitionState ->
+        Stack(modifier) {
+            PlantDetailsContent(
+                scrollerPosition = scrollerPosition,
+                toolbarShown = toolbarState.toolbarShown,
+                onToolbarShownUpdate = { newValue ->
+                    toolbarState = toolbarStateFromBoolean(newValue)
```

Comment:
I dislike having two versions of this, the enum and a boolean. Can we use one? The enum seems clearer to me.